File: utils/python/primes.py
def newPrimeSieve(size):
    # Create prime sieve and populate to assume all numbers are prime, initially
    prime_sieve = [True] * size

    # 0 and 1 are not prime
    prime_sieve[0] = False
    prime_sieve[1] = False

    i = 2
    while i < size:
        for index in range(i*i, size, i):
            prime_sieve[index] = False

        # Update i to the next prime exposed by the sieve
        i += 1
        while i < size and not prime_sieve[i]:
            i += 1
    
    return prime_sieve

def getPrimesLessThan(size):
    primeSieve = newPrimeSieve(size)

    primes = []

    for i in range(size):
        if primeSieve[i]:
            primes.append(i)
    
    return primes

File: utils/typescript/primes.ts
export function NewPrimeSieve(size: number): boolean[] {
    // Create prime sieve and populate to assume all numbers are prime, initially
    const prime_sieve: boolean[] = new Array<boolean>(size).fill(true);

    // 0 and 1 are not prime
    prime_sieve[0] = false
    prime_sieve[1] = false

    let i = 2
    while (i < size) {
        for (let index = i * i; index < size; index += i) {
            prime_sieve[index] = false
        }

        // Update i to the next prime exposed by the sieve
        do {
            i++
        } while (i < size && !prime_sieve[i])
    }

    return prime_sieve
}

export function GetPrimesLessThan(size: number): number[] {
    let primeSieve = NewPrimeSieve(size)

    let primes: number[] = []
    
    let i = 0
    for (const isPrime of primeSieve) {
        if (isPrime) {
            primes.push(i)
        }
        i++
    }

    return primes
}